src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,222 @@
+        return Optional.ofNullable(currentCar);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllCarsQuery = "SELECT cars.id AS car_id, model, "
+                + "manufacturers.id AS manufacturer_id,"
+                + " manufacturers.name AS manufacturer_name, manufacturers.country "
+                + "AS manufacturer_country FROM cars JOIN manufacturers "
+                + "ON cars.manufacturer_id = manufacturers.id WHERE cars.deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement getAllCarsStatement = connection.createStatement()) {
+            ResultSet dataFromDB = getAllCarsStatement.executeQuery(getAllCarsQuery);
+            while (dataFromDB.next()) {
+                cars.add(parseCarByDataFromDB(dataFromDB));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't select all cars from DB", e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriversForCurrentCar(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCarQuery = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarStatement = connection
+                        .prepareStatement(updateCarQuery)) {
+            updateCarStatement.setString(1, car.getModel());
+            updateCarStatement.setLong(2, car.getManufacturer().getId());
+            updateCarStatement.setLong(3, car.getId());
+            updateCarStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update Car object by input data: " + car, e);
+        }
+        deleteAllRelationBetweenCurrentCarAndDrivers(car.getId());
+        if (car.getDrivers() != null) {
+            insertDrivers(car);
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteCarQuery = "UPDATE cars SET deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteCarStatement = connection
+                        .prepareStatement(deleteCarQuery)) {
+            deleteCarStatement.setLong(1, id);
+            deleteCarStatement.executeUpdate();
+            return deleteCarStatement.executeUpdate() >= 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete data by input ID: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverID) {
+        String getAllByDriverQuery = "SELECT cars.id AS car_id FROM cars_drivers "
+                + "JOIN cars ON cars_drivers.cars_id = cars.id WHERE drivers_id = ?";
```

Comment:
same here. let's not fetch car id if driver is deleted

Reply:
added